src/runtime/providers/netlify.ts

```diff
@@ -1,50 +1,9 @@
-import { joinURL } from 'ufo'
-import type { ProviderGetImage } from '../../types'
-import { createOperationsGenerator } from '#image'
+import { getImage as largeMediaGetImage, operationsGenerator as largeMediaOptionsGenerator } from './netlifyLargeMedia'
+import { getImage as cdnGetImage, operationsGenerator as cdnOptionsGenerator } from './netlifyImageCdn'
 
-export const operationsGenerator = createOperationsGenerator({
-  keyMap: {
-    height: 'h',
-    fit: 'nf_resize',
-    width: 'w'
-  },
-  valueMap: {
-    fit: {
-      fill: 'smartcrop',
-      contain: 'fit'
-    }
-  },
-  joinWith: '&',
-  formatter: (key, value) => `${key}=${value}`
-})
+// If the site has the deprecated Netlify Large Media enabled we will use the old
+// Large Media adapter. Otherwise, we will use the Netlify Image CDN
+const hasLargeMedia = Boolean(process.env.NETLIFY_LFS_ORIGIN_URL)
```

Comment:
If `NETLIFY_LFS_ORIGIN_URL` is available in the build-time environment it would be worth doing this in the build/module scope rather than in the runtime. That way we avoid bundling both implementations.